### PR TITLE
DT-2: Move cloud authentication to cloud dashboard only

### PR DESCRIPTION
**Pull request by erickpintor**
This quite changes the original behaviour as such:

**Base dashboard**:
- Check for credentials in the session store
- If has credentials, skip the login form
- If no credentials, request endpoint and password and save them into the session store

**Cloud Dashboard**:
- Check if there is a website cookie with an auth key
- If have the cookie, calls website API to validate the user
- If has no cookie, redirect user to website login

_Cloud dashboard will never show the login form._

### CODE REVIEW

**Comment by jchris** [comment]
Does this support the website being able to offer the 1st behavior so that you can use it to connect to arbitrary endpoints or will the website dashboard be pinned to cloud?

**Comment by erickpintor** [comment]
The website dashboard would be pinned for cloud only. I've never managed to connect our cloud dashboard to anything but cloud due to CORS restriction. Do you have thoughts on that?

File: packages/dashboard-cloud/src/authentication/cloud-login.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+const CLOUD_COOKIE = "dashboard"
+const AUTH_API_TIMEOUT = 60000
+
+Events.listen("@@authentication/user-logged-out", () => {
+  Cookies.remove(CLOUD_COOKIE)
+})
+
+const parseResponse = (res) => {
+  if (res.ok) return res.json()
+  throw new Error("Unespected error while loggin into FaunaDB Cloud. Please, try again later.")
```
Comment (eaceaser) [inline packages/dashboard-cloud/src/authentication/cloud-login.js]: couple typos. Should be:

> Unexpected error while logging into FaunaDB Cloud.

I'm also up in the air about saying "Please try again later" - Should we say "please try again or check our status page for more info"?

**Comment by eaceaser** [comment]
@jchris @erickpintor Yeah, considering that the cloud dashboard can only talk to HTTPS endpoints, we can't guarantee any sort of solid experience when using the cloud-hosted dashboard with non-cloud endpoints. I think our effort to embed the dashboard in the dev jar / enterprise release will cover this case.

File: packages/dashboard-cloud/src/authentication/cloud-login.js (hunk 1 above)
Comment (erickpintor) [inline packages/dashboard-cloud/src/authentication/cloud-login.js]: Let's just remove the mention to "try again"... there is no evidence that it's a recoverable upon retry.

Comment (eaceaser) [inline packages/dashboard-cloud/src/authentication/cloud-login.js]: I do think we should add a link to the status page

File: packages/dashboard-cloud/src/authentication/auto-login.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+export default class AutoCloudLogin extends Component {
+  constructor(props) {
+    super(props)
+    this.state = { message: "Logging in..." }
+  }
+
+  componentDidMount() {
+    Events.listen("@@authentication/user-logged-out", () => {
+      this.setState({ message: "Loggin out..." }, () => {
```
Comment (eaceaser) [inline packages/dashboard-cloud/src/authentication/auto-login.js]: Typo: Loggin should be Logging

**Pull request merged**